### ISSUE TITLE
Can SACAD combine fragmented high resolution cover from amazon?

### ISSUE THREAD

**Issue by baoshan** [desbma/sacad#1]
Great project!

Can SACAD combine and output the highest resolution image from amazon as the below web application?

The highest resolution image is sliced into `3×3` small images.

http://www.amazon.com/gp/product/images/B004GP9GKA

**Comment by desbma** [comment]
Hi,

Currently SACAD does not extract and reassemble those images.

It seems they appear only for digital downloads, which are filtered out by the generated search URL :
* Manual search : http://www.amazon.com/s/ref=nb_sb_noss?url=search-alias%3Daps&field-keywords=art+tatum+live+at+the+shrine
* SACAD generated URL : http://www.amazon.com/gp/search?search-alias=popular&field-artist=art+tatum&field-title=live+at+the+shrine&sort=relevancerank

**Comment by desbma** [comment]
In your example the base image is 420x420, and when zooming there are 4 sliced images for a total resolution of 700x700. 

A search on Amazon for 'Art Tatum'  'Live at the Shrine' with SACAD currently returns two 492x500 images. So this would give potential additional results with 700 pixels square over 500 currently for the Amazon source only.

Not worth the effort in my opinion.

However if you find examples of URL where this method allows getting images on Amazon with a significantly larger size, I would be open to add support for it.

**Comment by baoshan** [comment]
Thanks for your experiment.

When inspecting the same digital release, I find the maximum resolution given by amazon from the above link is 1400 by 1400. Sliced into 4x4 = 16 small images.

![screen shot 2015-03-31 at 11 06 29 am](https://cloud.githubusercontent.com/assets/1096182/6911665/261d65f8-d796-11e4-88c4-8cf57b05af4b.png)

Compared to the standard resolution (which sometimes is a scanned version, not truly digital), I hope you could give the proposal a consideration when you have time.

I'm bad at Python. But when I need the feature in some months, I'll have a try to help when you haven't figured out enough time to do this.

Thanks.

**Comment by desbma** [comment]
For the URL you gave above I'm only getting 4 images  for a total resolution of 700x700, so they must do some differentiation based on browser or geographic origin.

What browser are you using ?

If there is a way to get 1400x1400 images instead of 500x500, I'll definitely have a look and try to implement something.

**Comment by baoshan** [comment]
I'm browsing using Safari 8.0.4, Chrome 41.0.2272, and FireFox 36.0.1. Tried on both a retina iMac and a non-retina external display.

All of the combinations request 4×4 slices. Sorry, I have no idea why you're getting 2×2 slices.

`1429 × 1417` or `1417 × 1417` is usually seen on label provided high-res cover arts. Exactly at 300dpi. Enough for archiving usage.

**Comment by desbma** [comment]
Can you please paste here the URL of two of the subimages ?

**Comment by baoshan** [comment]
```
http://z2-ec2.images-amazon.com/R/1/a=B004GP9GKA+c=A17SFUTIVB227Z+d=_SCR(3,0,0)_+o=01+s=RMTILE+va=MAIN+ve=363729657+e=.jpg
http://z2-ec2.images-amazon.com/R/1/a=B004GP9GKA+c=A17SFUTIVB227Z+d=_SCR(3,0,1)_+o=01+s=RMTILE+va=MAIN+ve=363729657+e=.jpg
http://z2-ec2.images-amazon.com/R/1/a=B004GP9GKA+c=A17SFUTIVB227Z+d=_SCR(3,0,2)_+o=01+s=RMTILE+va=MAIN+ve=363729657+e=.jpg
http://z2-ec2.images-amazon.com/R/1/a=B004GP9GKA+c=A17SFUTIVB227Z+d=_SCR(3,0,3)_+o=01+s=RMTILE+va=MAIN+ve=363729657+e=.jpg
http://z2-ec2.images-amazon.com/R/1/a=B004GP9GKA+c=A17SFUTIVB227Z+d=_SCR(3,1,0)_+o=01+s=RMTILE+va=MAIN+ve=363729657+e=.jpg
...
http://z2-ec2.images-amazon.com/R/1/a=B004GP9GKA+c=A17SFUTIVB227Z+d=_SCR(3,3,3)_+o=01+s=RMTILE+va=MAIN+ve=363729657+e=.jpg
```

**Comment by desbma** [comment]
Thanks,

I had a look and it's possible to automate the grabbing and joining of those images.

It's a bit more involved than the current Amazon code because the "A17SFUTIVB227Z" part of the URL is generated from JavaScript which SACAD does not interpret.

Also some albums don't have 4x4 images but only 3x3 or 2x2 and there is no way to know except by trying and see if it succeeds.

I will implement it in a new source soon : `AmazonDigital` (vs `AmazonCd` for the classic one).

**Comment by desbma** [comment]
Added in 7378de6031a2b9cea9e0e9e5817ddc7f2c3ec8ad and available in [v1.3.0](https://github.com/desbma/sacad/releases/tag/1.3.0)

    $ sacad 'art tatum' 'live at the shrine' 1400 /tmp/a.jpg 
    Got 0 relevant (0 excluded) results from source 'LastFmCoverSource'
    Got 0 relevant (0 excluded) results from source 'CoverLibCoverSource'
    Got 0 relevant (2 excluded) results from source 'AmazonCdCoverSource'
    Got 1 relevant (0 excluded) results from source 'AmazonDigitalCoverSource'
    Got 14 relevant (6 excluded) results from source 'GoogleImagesWebScrapeCoverSource'
    No reference result found
    Downloading cover 'http://z2-ec2.images-amazon.com/R/1/a=B004GP7J72+c=A17SFUTIVB227Z+d=_SCR%283,0,0%29_=.jpg' (part 1/16)...
    Crunching JPEG image...
    Downloading cover 'http://z2-ec2.images-amazon.com/R/1/a=B004GP7J72+c=A17SFUTIVB227Z+d=_SCR%283,0,1%29_=.jpg' (part 2/16)...
    Crunching JPEG image...
    Downloading cover 'http://z2-ec2.images-amazon.com/R/1/a=B004GP7J72+c=A17SFUTIVB227Z+d=_SCR%283,0,2%29_=.jpg' (part 3/16)...
    Crunching JPEG image...
    Downloading cover 'http://z2-ec2.images-amazon.com/R/1/a=B004GP7J72+c=A17SFUTIVB227Z+d=_SCR%283,0,3%29_=.jpg' (part 4/16)...
    Crunching JPEG image...
    Downloading cover 'http://z2-ec2.images-amazon.com/R/1/a=B004GP7J72+c=A17SFUTIVB227Z+d=_SCR%283,1,0%29_=.jpg' (part 5/16)...
    Crunching JPEG image...
    Downloading cover 'http://z2-ec2.images-amazon.com/R/1/a=B004GP7J72+c=A17SFUTIVB227Z+d=_SCR%283,1,1%29_=.jpg' (part 6/16)...
    Crunching JPEG image...
    Downloading cover 'http://z2-ec2.images-amazon.com/R/1/a=B004GP7J72+c=A17SFUTIVB227Z+d=_SCR%283,1,2%29_=.jpg' (part 7/16)...
    Crunching JPEG image...
    Downloading cover 'http://z2-ec2.images-amazon.com/R/1/a=B004GP7J72+c=A17SFUTIVB227Z+d=_SCR%283,1,3%29_=.jpg' (part 8/16)...
    Crunching JPEG image...
    Downloading cover 'http://z2-ec2.images-amazon.com/R/1/a=B004GP7J72+c=A17SFUTIVB227Z+d=_SCR%283,2,0%29_=.jpg' (part 9/16)...
    Crunching JPEG image...
    Downloading cover 'http://z2-ec2.images-amazon.com/R/1/a=B004GP7J72+c=A17SFUTIVB227Z+d=_SCR%283,2,1%29_=.jpg' (part 10/16)...
    Crunching JPEG image...
    Downloading cover 'http://z2-ec2.images-amazon.com/R/1/a=B004GP7J72+c=A17SFUTIVB227Z+d=_SCR%283,2,2%29_=.jpg' (part 11/16)...
    Crunching JPEG image...
    Downloading cover 'http://z2-ec2.images-amazon.com/R/1/a=B004GP7J72+c=A17SFUTIVB227Z+d=_SCR%283,2,3%29_=.jpg' (part 12/16)...
    Crunching JPEG image...
    Downloading cover 'http://z2-ec2.images-amazon.com/R/1/a=B004GP7J72+c=A17SFUTIVB227Z+d=_SCR%283,3,0%29_=.jpg' (part 13/16)...
    Crunching JPEG image...
    Downloading cover 'http://z2-ec2.images-amazon.com/R/1/a=B004GP7J72+c=A17SFUTIVB227Z+d=_SCR%283,3,1%29_=.jpg' (part 14/16)...
    Crunching JPEG image...
    Downloading cover 'http://z2-ec2.images-amazon.com/R/1/a=B004GP7J72+c=A17SFUTIVB227Z+d=_SCR%283,3,2%29_=.jpg' (part 15/16)...
    Crunching JPEG image...
    Downloading cover 'http://z2-ec2.images-amazon.com/R/1/a=B004GP7J72+c=A17SFUTIVB227Z+d=_SCR%283,3,3%29_=.jpg' (part 16/16)...
    Crunching JPEG image...
    Joining 16 images...
    Crunching JPEG image...
